module/Category/src/Controller/CategoryController.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Category\Controller;
+
+use Zend\Mvc\Controller\AbstractActionController;
+use Zend\View\Model\ViewModel;
+use Doctrine\ORM\EntityManager;
+use Category\Repository\CategoryRepository;
+
+class CategoryController extends AbstractActionController
+{
+    private $entityManager;
+
+    public function __construct($serviceManager)
+    {
+       $this->entityManager = $serviceManager->get(EntityManager::class);
+    }
+
+    public function indexAction()
+    {
+        $categories = new CategoryRepository($this->entityManager);
+        $data = $categories->findAll();
+        return new ViewModel(['categories' => $data]);
+    }
+
+    public function addAction()
+    {
+        return new ViewModel();
+    }
+
+    public function addAjaxAction()
+    {
+        if($this->getRequest()->isPost()) {
+           $category = new CategoryRepository($this->entityManager);
```

Comment:
See above re: using service instead of business logic directly inside controller.